### PR TITLE
add documentation of OCSP_basic_verify()

### PR DESCRIPTION
**Pull request by DDvO**
The function `OCSP_basic_verify()` so far appears undocumented.
In order to be able to use it properly, I needed to have a very close look at its implementation.
This PR attempts to fill the gap by reporting what the function does, hopefully correctly and at the right level of detail.

- [x] documentation is added or updated

### CODE REVIEW

File: doc/man3/OCSP_resp_find_status.pod [hunk 1]
```diff
@@ -100,6 +104,23 @@ OCSP_single_get0_status(). If B<sec> is non-zero it indicates how many seconds
 leeway should be allowed in the check. If B<maxsec> is positive it indicates
 the maximum age of B<thisupd> in seconds.
 
+OCSP_basic_verify() checks that the basic response message B<bs> is correctly
+signed (unless the B<OCSP_NOSIGS> flag is set) and that the signer certificate
+can be validated (unless the <OCSP_NOVERIFY> flag is set). It takes B<st> as the
+trusted store and B<certs> as a set of untrusted intermediate certificates.
+The function tries to find the signer certificate of the response in B<cert>.
```
Comment (mattcaswell) [inline doc/man3/OCSP_resp_find_status.pod]: typo: cert*s*

File: doc/man3/OCSP_resp_find_status.pod [hunk 2]
```diff
@@ -100,6 +104,23 @@ OCSP_single_get0_status(). If B<sec> is non-zero it indicates how many seconds
...
+certificates the responder may have included in B<bs>.
+It then validates the signer certificate unless it was found in B<certs>
+and the B<OCSP_TRUSTOTHER> flag is set. Otherwise both sources of untrusted
+certificates are used for constructing the validation path for the signer
+certificate unless the B<OCSP_NOCHAIN> flag is set. After successful
+path validation the function immediately yields a positive result if the
+B<OCSP_NOCHECKS> flag is set. Else it verifies that the signer certificate
+meets the OCSP issuer criteria using B<ocsp_check_issuer()>. If the latter
+gives a negative result and the flag B<OCSP_NOEXPLICIT> is not set, the
+function checks for explicit trust for OCSP signing in the root CA certificate.
```
Comment (mattcaswell) [inline doc/man3/OCSP_resp_find_status.pod]: I might be tempted to reformulate this as a description of what happens in the default "no flags" case, followed by a list of all the flags explaining how they alter its operation.

File: doc/man3/OCSP_resp_find_status.pod [hunk 3]
```diff
@@ -100,6 +104,23 @@ OCSP_single_get0_status(). If B<sec> is non-zero it indicates how many seconds
...
+The function tries to find the signer certificate of the response in B<cert>.
+If not found and the B<OCSP_NOINTERN> flag is not set, it also tries any
+certificates the responder may have included in B<bs>.
+It then validates the signer certificate unless it was found in B<certs>
+and the B<OCSP_TRUSTOTHER> flag is set. Otherwise both sources of untrusted
+certificates are used for constructing the validation path for the signer
+certificate unless the B<OCSP_NOCHAIN> flag is set. After successful
+path validation the function immediately yields a positive result if the
+B<OCSP_NOCHECKS> flag is set. Else it verifies that the signer certificate
+meets the OCSP issuer criteria using B<ocsp_check_issuer()>. If the latter
```
Comment (mattcaswell) [inline doc/man3/OCSP_resp_find_status.pod]: `ocsp_check_issuer` is an internal static function, so it probably shouldn't appear in the public API docs.

File: doc/man3/OCSP_resp_find_status.pod (hunk 1 above)
Comment (DDvO) [reply doc/man3/OCSP_resp_find_status.pod]: Corrected.

File: doc/man3/OCSP_resp_find_status.pod (hunk 2 above)
Comment (DDvO) [reply doc/man3/OCSP_resp_find_status.pod]: I see your point and tried this. Doing so would require repeating part of the description of the "no flags" behavior in order to set the context for describing the effect of the respective flags. So I gave up on this as I felt the overall text would become more complicated and significantly longer. Still I was able to streamline the text a bit for better readability.

File: doc/man3/OCSP_resp_find_status.pod (hunk 3 above)
Comment (DDvO) [reply doc/man3/OCSP_resp_find_status.pod]: Good point - I removed the reference to the internal function.

**Comment by DDvO** [comment]
@mattcaswell, WDYT about the revised text?

**Comment by levitte** [comment]
This looks ready.  Should this be backported to 1.1.0?

**Comment by richsalz** [comment]
the files moved so a vanilla cherry-pick won't work.  My view is that not all bug fixes need to be backported; I know others disagree.

**Comment by levitte** [comment]
Sounds like the lazy argument to me.  It's not really *that* hard.  I'm willing to do the work.

**Comment by richsalz** [comment]
If time/resources were infinite I might agree with you.  But everything needs to be prioritized hence my opinions.

**Comment by levitte** [comment]
It takes me less time to do it than it does to argue about it with you... Just sayin'

**Comment by richsalz** [comment]
I didn't argue to not do it.  I just think we need to have a mindset that not all bugfixes automatically get backported.  Just sayin'

**Comment by levitte** [comment]
Ok, your general opinion noted.  The question regarding *this* PR still stands

**Comment by richsalz** [comment]
reconfirm to backport.

**Comment by levitte** [comment]
Merged.

master / 1.1.1:
b8c32081e02b7008a90d878eccce46da256dfe86 add documentation for OCSP_basic_verify()

1.1.0:
2b451311ee5973907da650ed25393130310c3841 add documentation for OCSP_basic_verify()

**Comment by levitte** [comment]
(we don't have any OCSP documentation on 1.0.2, so no backport that far)

**Pull request closed without merging**